### PR TITLE
Redirect FREE_TEST_PLAN to paywall + allow trial

### PR DESCRIPTION
**Pull request by aubin-tchoi**
## Description

- Close [#1756](https://github.com/dust-tt/tasks/issues/1756).
- Redirect FREE_TEST_PLAN to paywall.
- Allow 14 days of trials for FREE_TEST_PLAN.
- Update the paywall messages for them ("activate" billing instead of "resume")
- Set `canUseProduct` to false for FREE_TEST_PLAN.

Tested by upgrading my workspace to FREE_TEST_PLAN + clear cookies, correctly redirected to the paywall.

## Risk

Low

## Deploy Plan

- Run `20241205_free_test_plan_cant_use_product.sql`.
- Deploy front.

### CODE REVIEW

File: front/lib/iam/session.ts [hunk 1]
```diff
@@ -135,7 +136,8 @@ export function makeGetServerSidePropsRequirementsWrapper<
 
       if (
         requireCanUseProduct &&
-        !auth?.subscription()?.plan.limits.canUseProduct
+        (!auth?.subscription()?.plan.limits.canUseProduct ||
```
Comment (PopDaph) [inline front/lib/iam/session.ts:139]: I completely forgot about the canUseProduct, I think we should remove this change and update the plan in db instead to set it to false for this plan. WDYT?

**Review by PopDaph — APPROVED** [review]
Just one comment, the rest looks good to me, thank you! 🙏🏻

File: front/lib/iam/session.ts (hunk 1 above)
Comment (aubin-tchoi) [reply front/lib/iam/session.ts:139]: Sounds good, having `canUseProduct` to true is indeed weird

**Review by PopDaph — APPROVED** [review]

File: front/lib/plans/stripe.ts [hunk 2]
```diff
@@ -117,7 +120,7 @@ export const createProPlanCheckoutSession = async ({
       workspaceId: owner.id,
     },
   });
-  if (existingSubscription) {
+  if (existingSubscription && !isOldFreePlan(existingSubscription.plan.code)) {
```
Comment (flvndvd) [inline front/lib/plans/stripe.ts:123]: Looks like the subscription's plan does not exist anymore in prod.

Comment (flvndvd) [reply front/lib/plans/stripe.ts:123]: If you need the plan information you should include it when fetching the subscription (line 118).

**Pull request merged**